chef/automate
# API to fetch the organisation users from automate DB

### :nut_and_bolt: Description: What code changed, and why?
List the users in an organisation from automate DB



### :chains: Related Resources
https://chefio.atlassian.net/jira/software/c/projects/STALWART/boards/361?modal=detail&selectedIssue=STALWART-45

### :+1: Definition of Done
I have added API to list the users in an organisation from automate DB

### :athletic_shoe: How to Build and Test the Change
Steps to build:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
Steps to test:

1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
2. Add the chef-server test environment credentials and extract the SERVER IDs

Request

```curl -sSkH "api-token: $(get_admin_token)" 'https://a2-dev.test/api/v0/infra/servers/SERVER_ID/org/ORG_ID/automateinfraorgusers'```

Response

```
{
    "users": [
        {
            "user_id": 2,
            "server_id": "ec2",
            "org_id": "demoorg",
            "infra_server_username": "test-new-user",
            "first_name": "",
            "last_name": "",
            "email_id": "test-new@email.com",
            "middle_name": "",
            "display_name": "test-new-user",
            "connector": "LOCAL",
            "automate_user_id": "test-new-user",
            "is_admin": false
        },
        {
            "user_id": 3,
            "server_id": "ec2",
            "org_id": "demoorg",
            "infra_server_username": "kallol",
            "first_name": "Kallol",
            "last_name": "Roy",
            "email_id": "kallol.roy@progress.com",
            "middle_name": "",
            "display_name": "Kallol Roy",
            "connector": "LOCAL",
            "automate_user_id": "kallol",
            "is_admin": true
        }
    ]
}
```

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
